### PR TITLE
add userId to socket requests

### PR DESCRIPTION
**Pull request by bobobob22**

### CODE REVIEW

File: packages/signaling-server/src/index.ts [hunk 1]
```diff
@@ -28,31 +32,59 @@ void (async () => {
...
+
   // @ts-ignore
   const io = await socket(server, {
     cors: {
       origin: '*',
     },
   })
 
+  // fetch
+  const PUB_KEY_URL = 'https://api.staging.tt.fm/auth/pub-key'
```
Comment (kowalski) [inline packages/signaling-server/src/index.ts:44]: get this from env;
make a note to add this env to TaskDefinition in infrastracture repo

File: packages/roomSync/src/store/initState/initState.saga.ts [hunk 2]
```diff
@@ -41,6 +42,8 @@ export function* onGetInitStateMessage(action: InitStateActions['onGetInitStateM
     requestingPeer: action.payload.from,
   }
 
+  console.log('3')
```
Comment (kowalski) [inline packages/roomSync/src/store/initState/initState.saga.ts]: leftover

File: packages/signaling-server/src/helpers/configureListenersForRoomBroadcastSocketIo.ts [hunk 3]
```diff
@@ -39,5 +40,5 @@ export const configureListenersForRoomBroadcast = (roomBroadcast: RoomBroadcast,
   roomBroadcast.addListenerToChatRoomBroadcast('peer:disconnected', (peerId: string) =>
     io.emit('userDisconnected', peerId),
   )
-  roomBroadcast.addListenerToChatRoomBroadcast('removeAvatar', (peerId: string) => io.emit('removeAvatar', peerId))
+  roomBroadcast.addListenerToChatRoomBroadcast('removeAvatar', (data: IMessage & IDisconnectPeer) => io.emit('removeAvatar', data.userPeerId))
```
Comment (kowalski) [inline packages/signaling-server/src/helpers/configureListenersForRoomBroadcastSocketIo.ts:43]: we are not passing `userId` in this message?

File: packages/signaling-server/src/index.ts (hunk 1 above)
Comment (kowalski) [inline packages/signaling-server/src/index.ts:44]: take from env, add in TaskDefinition in infrastracture repo

File: packages/signaling-server/src/index.ts [hunk 4]
```diff
@@ -28,31 +32,59 @@ void (async () => {
...
+  }))
+
+
   const store = await configureStore(roomBroadcast)
   store.dispatch(peerIdActions.setPeerId(peerId))
   configureListenersForRoomBroadcast(roomBroadcast, store, io)
 
   io.on('connection', function (socket) {
-    const connectedSocketId = socket.client.id
+    const connectedSocketId = socket.handshake.user.id
```
Comment (kowalski) [inline packages/signaling-server/src/index.ts:73]: this is not socket id. This is userId. This variable name is confusing

File: packages/webapp/src/containers/Dancefloor.container.tsx [hunk 5]
```diff
@@ -20,9 +20,22 @@ export const DancefloorContainer: FC = () => {
   const approvedPeerIds = useSelector(satisfactionMeterSelectors.approvedPeerIds)
   const amICurrentlyPlayingDj = useSelector(amICurrentlyPlayingDjSelector)
   const userPeerJson = useSelector(peerIdSelectors.peerIdJson)
+  const userId = useSelector(userSelectors.userId)
 
   const dispatch = useDispatch()
 
+  const dispatchSendSatisfaction = (choice: string): void => {
```
Comment (kowalski) [inline packages/webapp/src/containers/Dancefloor.container.tsx:27]: useCallbacks with correct dependencies is welcome

File: packages/webapp/src/store/user/user.saga.test.ts [hunk 6]
```diff
@@ -398,10 +398,10 @@ describe('users', () => {
       await expectSaga(
         sendSatisfactionHandler,
         roomBroadcast,
-        broadcastActions.sendSatisfaction({ choice: 'approve', userPeerId: '2' }),
+        broadcastActions.sendSatisfaction({ choice: 'approve', userId: '2', userPeerId: '2' }),
```
Comment (kowalski) [inline packages/webapp/src/store/user/user.saga.test.ts:406]: its passed as string here `'2'`. Isnt' userId `number` ?

File: packages/webapp/src/store/user/user.saga.ts [hunk 7]
```diff
@@ -75,7 +82,17 @@ export function* sendSatisfactionHandler(
...
-  yield apply(roomBroadcast, roomBroadcast.sendSatisfaction, [action.payload.choice, action.payload.userPeerId])
+  const userId = yield* select(userSelectors.userId)
+
+  if (!userId) {
+    return
+  }
+
+  yield apply(roomBroadcast, roomBroadcast.sendSatisfaction, [
+    action.payload.choice,
+    userId.toString(),
```
Comment (kowalski) [inline packages/webapp/src/store/user/user.saga.ts]: action should get userId as number imo. No need for toString()

**Review by kowalski — CHANGES_REQUESTED** [review]

File: packages/signaling-server/package.json [hunk 8]
```diff
@@ -10,16 +10,14 @@
...
+  "license": "ISC",
+  "dependencies": {
     "@hapi/hapi": "^20.0.0",
     "@hapi/inert": "^6.0.2",
-    "menoetius": "0.0.2",
-    "prom-client": "^13.0.0",
+    "@turntable/libp2p-webrtc-star": "*",
     "@turntable/roomSync": "*",
-    "minimist": "^1.2.5",
+    "@types/passport-jwt": "^3.0.3",
```
Comment (kowalski) [inline packages/signaling-server/package.json:20]: this one should be in devDependencies

File: packages/roomSync/src/store/initState/initState.saga.ts (hunk 2 above)
Comment (bobobob22) [reply packages/roomSync/src/store/initState/initState.saga.ts]: ok

File: packages/signaling-server/src/index.ts (hunk 4 above)
Comment (bobobob22) [reply packages/signaling-server/src/index.ts]: ok

File: packages/webapp/src/store/user/user.saga.test.ts (hunk 6 above)
Comment (bobobob22) [reply packages/webapp/src/store/user/user.saga.test.ts]: ok

File: packages/webapp/src/store/user/user.saga.ts (hunk 7 above)
Comment (bobobob22) [reply packages/webapp/src/store/user/user.saga.ts]: ok

File: packages/signaling-server/src/index.ts (hunk 1 above)
Comment (bobobob22) [reply packages/signaling-server/src/index.ts:44]: ok

File: packages/signaling-server/package.json (hunk 8 above)
Comment (bobobob22) [reply packages/signaling-server/package.json:20]: ok

**Pull request merged**